string name
string description
string repo_id
int64 max_steps